### ISSUE TITLE
[AI/ML] Build Baseline ICDR 0–4 DR Classification Model

### ISSUE THREAD

**Issue by Devrajj-Singh** [Devrajj-Singh/NeuroVista-DR#4]
### Objective

Develop the first working AI model capable of classifying retinal fundus images into the five ICDR DR severity levels.

This is the **core AI component of the initial prototype**.

### Initial Model

Start with:

**EfficientNet-B0 using transfer learning**

Do not spend the initial prototype period comparing many architectures. The first goal is a reproducible working model that can perform inference on a new fundus image.

### Input

```text
RGB fundus image
```

after the defined preprocessing pipeline.

### Output

```text
Predicted DR Grade
Class probabilities
Confidence
```

Example:

```text
Predicted Grade: 2
Severity: Moderate NPDR
Confidence: 0.87
```

### Training Tasks

- [ ] Load pretrained EfficientNet-B0
- [ ] Replace classification head
- [ ] Configure 5-class output
- [ ] Load processed dataset
- [ ] Apply preprocessing
- [ ] Apply training augmentation
- [ ] Configure loss
- [ ] Configure optimizer
- [ ] Train baseline
- [ ] Save best checkpoint
- [ ] Evaluate on validation set
- [ ] Evaluate on held-out test set
- [ ] Generate confusion matrix
- [ ] Calculate accuracy
- [ ] Calculate precision
- [ ] Calculate recall/sensitivity
- [ ] Calculate specificity where applicable
- [ ] Calculate F1-score
- [ ] Record training configuration

### Important Evaluation Rule

Do not claim SIH targets have been achieved unless actual test results demonstrate them.

Report:

```text
Target:
>90% sensitivity
>85% specificity

Actual:
<measured result>
```

### Model Documentation

Document:

```text
Model:
Dataset:
Input size:
Preprocessing:
Augmentation:
Optimizer:
Learning rate:
Batch size:
Epochs:
Loss:
Best checkpoint:
Validation metrics:
Test metrics:
```

### Acceptance Criteria

Given a new fundus image, the inference pipeline must return:

```text
grade ∈ {0,1,2,3,4}
```

along with usable confidence/probability information.

The inference function must be callable by the backend/integration layer.
